docs/guides/aws-private-link-workspace.md

```diff
@@ -0,0 +1,261 @@
+)
+workspace_subnet_2 = (
+ element(split(",", var.existing_network_subnets), 2)
+)
+}
+```
+
+## Configure AWS objects
+The first step is to create the required AWS objects:
+- A subnet dedicated to your VPC endpoints
+- A security group dedicated to your VPC endpoints and satisfying required inbound/outbound TCP/HTTPS traffic rules on ports 443 and 6666, respectively.
+- Lastly, creation of the private access settings and workspace.
+
+```hcl
+resource "aws_subnet" "vpce" {
+  vpc_id     = var.vpc_id
+  cidr_block = var.vpce_cidr
+
+  tags = {
+    Name = "vpce subnet for workspace"
+  }
+}
+
+resource "aws_route_table" "my_vpc_private" {
+    vpc_id = var.vpc_id
+
+    tags = {
+        Name = "Local Route Table for Isolated Private Subnet"
+    }
+}
+
+resource "aws_route_table_association" "my_vpc_us_east_1a_private" {
+    subnet_id = aws_subnet.vpce.id
+    route_table_id = aws_route_table.my_vpc_private.id
+}
+```
+
+```hcl
+resource "aws_security_group" "vpce_sg" {
+ name        = "VPC endpoint security group"
+ description = "Security group shared with relay and workspace endpoints"
+ vpc_id      = var.vpc_id
+
+ ingress {
+   description      = "Inbound rules"
+   from_port        = 443
+   to_port          = 443
+   protocol         = "tcp"
+   cidr_blocks      = [var.vpc_cidr_block]
+ }
+
+ ingress {
+   description      = "Outbound rules"
+   from_port        = 6666
+   to_port          = 6666
+   protocol         = "tcp"
+   cidr_blocks      = [var.vpc_cidr_block]
+ }
+
+ egress {
+   from_port        = 0
+   to_port          = 0
+   protocol         = "-1"
+   cidr_blocks      = ["0.0.0.0/0"]
+   ipv6_cidr_blocks = ["::/0"]
+ }
+
+ tags = {
+   Name = "vpce_rules"
+ }
+}
+```
+
+```hcl
+resource "aws_vpc_endpoint" "workspace" {
+ vpc_id             = var.vpc_id
+ service_name       = var.workspace_vpce_service
+ vpc_endpoint_type  = "Interface"
+ security_group_ids = [aws_security_group.vpce_sg.id]
+ subnet_ids         = [aws_subnet.vpce.id]
+ // run terraform apply twice when configuring PrivateLink.
+ // Run 1 - comment the `private_dns_enabled` line
+ // Run 2 - uncomment the `private_dns_enabled` line
+ // private_dns_enabled = true
+ depends_on         = [aws_subnet.vpce]
+}
+
+resource "aws_vpc_endpoint" "relay" {
+ vpc_id             = var.vpc_id
+ service_name       = var.relay_vpce_service
+ vpc_endpoint_type  = "Interface"
+ security_group_ids = [aws_security_group.vpce_sg.id]
+ subnet_ids         = [aws_subnet.vpce.id]
+   // run terraform apply twice when configuring PrivateLink.
+ // Run 1 - comment the `private_dns_enabled` line
+ // Run 2 - uncomment the `private_dns_enabled` line
+ // private_dns_enabled = true
+ depends_on         = [aws_subnet.vpce]
+}
+
+resource "databricks_mws_vpc_endpoint" "workspace" {
+ provider            = databricks.mws
+ account_id          = var.databricks_account_id
+ aws_vpc_endpoint_id = aws_vpc_endpoint.workspace.id
+ vpc_endpoint_name   = "VPC Relay for ${var.vpc_id}"
```

Comment:
1) include prefix in names - there might be many endpoints.
2) `workspace` and `relay` endpoints have the same name now. fix it.

Reply:
Fixed.